cpp/test/sg/kmeans_test.cu

```diff
@@ -70,6 +70,10 @@ class KmeansTest : public ::testing::TestWithParam<KmeansInputs<T>> {
     allocate(d_labels, n_samples);
     allocate(d_labels_ref, n_samples);
     allocate(d_centroids, params.n_clusters * n_features);
+    allocate(d_sample_weight, n_samples);
```

Comment:
can you also add the test for the case when sample weights are not needed? This is just to make sure that we have coverage for both code paths.

Reply:
done